pyvo/dal/tap.py

```diff
@@ -821,7 +821,7 @@ def raise_if_error(self):
             if theres an error
         """
         if self.phase in {"ERROR", "ABORTED"}:
-            raise DALQueryError("Query Error", self.phase, self.url)
+            raise DALQueryError("Query Error: " + self.job.message, self.phase, self.url)
```

Comment:
Two thoughts here.

1) I couldn't convince myself quickly that `self.job.message` would have a value here.  Would it be worth checking for that first?

2) I have a general concern with the use of the `job` and `phase` properties here.  My real problem is with how those "properties" are defined, but I don't suggest any changes there for this PR.   Those "properties" are both written to do an `_update()` **every** time they are accessed, and `_update()` hits network at the TAP service job URL.  So for these two lines, we would do 3 GETs of the same information just to craft an error message.  That's not efficient and if the error is related to communication problems with the service, then this makes it worse.

One of the reasons I don't like these properties is that there's no good way to be sure it's safe to use the underscore version (i.e., `_job`, `_phase` which don't do `_updates()`).  In this case I think it would be safe to use the underscore versions on all 3 accesses, but it certain to be safe to leave the `self.phase` in the `if` and use `self._job...` and `self._phase` in the error call since the phase check would have called `_update()`.

Reply:
@tomdonaldson for 1:
```tap = pyvo.dal.TAPService("https://ws.cadc-ccda.hia-iha.nrc-cnrc.gc.ca/youcat")
     tap.run_async("select top 2 * from caom2.Observations")
     ```
Currently, the error is:
```
  File "/Users/adriand/Documents/work/github/pyvo/pyvo/dal/tap.py", line 824, in raise_if_error
    raise DALQueryError("Query Error ", self.url)
pyvo.dal.exceptions.DALQueryError: Query Error 
```

The the job error message:
```
pyvo.dal.exceptions.DALQueryError: Query Error: IllegalArgumentException:Table [ caom2.Observations ] is not found in TapSchema. Possible reasons: table does not exist or permission is denied.
```
which is an improvement.

Reply:
I think this might address #125 too

Reply:
No 2. has an issue now #365